### ISSUE TITLE
container exits(0) - Different lower_case_table_names settings for server ('2') and data dictionary ('0')

### ISSUE THREAD

**Issue by WouterDoornbos** [laradock/laradock#2732]
### Expected Behavior:
After installing Laradock and executing:` docker-compose up -d nginx mysql phpmyadmin` all the docker containers should run fine.
### Context information:
**Output of `docker version`**
```
Client: Docker Engine - Community
 Cloud integration  0.1.18
 Version:           19.03.13
 API version:       1.40
 Go version:        go1.13.15
 Git commit:        4484c46d9d
 Built:             Wed Sep 16 16:58:31 2020
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.13
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       4484c46d9d
  Built:            Wed Sep 16 17:07:04 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.3.7
  GitCommit:        8fba4e9a7d01810a393d5d25a3621dc101981175
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

**Output of `docker-compose version`**
```
docker-compose version 1.27.4, build 40524192
docker-py version: 4.3.1
CPython version: 3.7.7
OpenSSL version: OpenSSL 1.1.1g  21 Apr 2020
```

### Steps to reproduce the issue:
Clone the repo and run ` docker-compose up -d nginx mysql phpmyadmin` the MySQL exits with code (0)

When I try to start the container again I get the following message:

```
mysql_1                | 2020-10-01T10:19:52.124915Z 1 [Warning] [MY-012579] [InnoDB] fallocate(25, FALLOC_FL_PUNCH_HOLE | FALLOC_FL_KEEP_SIZE, 0, 16384) returned errno: 22
mysql_1                | 2020-10-01T10:19:52.143672Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
mysql_1                | 2020-10-01T10:19:52.159801Z 1 [ERROR] [MY-011087] [Server] Different lower_case_table_names settings for server ('2') and data dictionary ('0').
mysql_1                | 2020-10-01T10:19:52.160778Z 0 [ERROR] [MY-010020] [Server] Data Dictionary initialization failed.
mysql_1                | 2020-10-01T10:19:52.161607Z 0 [ERROR] [MY-010119] [Server] Aborting
mysql_1                | 2020-10-01T10:19:52.795909Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 8.0.21)  MySQL Community Server - GPL.
```

### Stacktrace & Additional info:

I tried adding `"command: --lower_case_table_names=0" `to my docker-compose file, but that does not seem to help.

MySQL crashes with:

```
mysql_1                | 2020-10-01 10:21:28+00:00 [ERROR] [Entrypoint]: mysqld failed while attempting to check config
mysql_1                |        command was: mysqld --lower_case_table_names=0 --verbose --help
mysql_1                |        2020-10-01T10:21:28.547104Z 0 [ERROR] [MY-010158] [Server] The server option 'lower_case_table_names' is configured to use case sensitive table names but the data directory is on a case-insensitive file system which is an unsupported combination. Please consider either using a case sensitive file system for your data directory or switching to a case-insensitive table name mode.
mysql_1                | 2020-10-01T10:21:28.549273Z 0 [ERROR] [MY-010119] [Server] Aborting
```

**Comment by marcoraddatz** [comment]
Had the same issue. After deleting the local DB everything worked again.

**Comment by talifhani** [comment]
> Had the same issue. After deleting the local DB everything worked again.

Please explain what you mean by delete local DB? Having the same issue :(

**Comment by talifhani** [comment]
Found the solution: https://stackoverflow.com/questions/64146845/mysql-not-starting-in-a-docker-container-on-macos-after-docker-update

**Comment by amaurer** [comment]
> Found the solution: https://stackoverflow.com/questions/64146845/mysql-not-starting-in-a-docker-container-on-macos-after-docker-update

Thanks, this saved me a lot of time.

**Comment by Maras0830** [comment]
Thans @talifhani ~

**Comment by radsssss** [comment]
My solution:

cmd

fsutil file setCaseSensitiveInfo "C:\Users\XXXXX\\.laradock\data\mysql"

then 

docker-compose up mysql

**Comment by pjotrsavitski** [comment]
That happened after an update to a newer version about a month ago, probably 2.4.x. It has either got a new option to deal with file sharing or it just got automatically enabled (use gRPC FUSE ...). Disabling that one will allow one to continue without issues. Check out the attached screenshot. This was true for Docker running on MacOS with default APFS file system being **case-insensitive**.

![Screenshot 2020-11-05 at 18 04 06](https://user-images.githubusercontent.com/518331/98265479-cde52e80-1f91-11eb-9d3d-57f1a9ff8168.png)

This boils down to case-sensitivity being suddenly changed compared to what it used to be before. You can even keep the new way with changing the setting the way it used to be, exporting the database, deleting the database, changing the setting and then importing the database.

**Comment by zabaala** [comment]
> That happened after an update to a newer version about a month ago, probably 2.4.x. It has either got a new option to deal with file sharing or it just got automatically enabled (use gRPC FUSE ...). Disabling that one will allow one to continue without issues. Check out the attached screenshot. This was true for Docker running on MacOS with default APFS file system being **case-insensitive**.
> 
> ![Screenshot 2020-11-05 at 18 04 06](https://user-images.githubusercontent.com/518331/98265479-cde52e80-1f91-11eb-9d3d-57f1a9ff8168.png)
> 
> This boils down to case-sensitivity being suddenly changed compared to what it used to be before. You can even keep the new way with changing the setting the way it used to be, exporting the database, deleting the database, changing the setting and then importing the database.

Thanks, @pjotrsavitski. This solution work fine for me on Docker Desktop 2.5.0.0 (49427) and Mysql 8.0.2 image.

**Comment by qingdi** [comment]
![image](https://user-images.githubusercontent.com/5287422/98753222-cd6bee00-23fe-11eb-90d5-db15bb2dddd0.png)

**Comment by yuhal** [comment]
- Mysql 8.0.21
- Docker Desktop 2.5.0.1
> The above is the version information. Open Docker Desktop, find `Experimental Features`, close `Use gRPC FUSE for file sharing`, and finally solve this problem.
Thanks, @qingdi.

**Comment by rafa1944** [comment]
You saved my day guys! THANKS

**Comment by Shaykhnazar** [comment]
> My solution:
> 
> cmd
> 
> fsutil file setCaseSensitiveInfo "C:\Users\XXXXX\.laradock\data\mysql"
> 
> then
> 
> docker-compose up mysql

Thanks a lot works for windows 10. I have the same issue.

**Comment by stardusteight-d4c** [comment]
In my case I got the error:

```
mysql  | 2023-05-04T18:28:09.899430Z 1 [ERROR] [MY-011087] [Server] Different lower_case_table_names settings for server ('0') and data dictionary ('2').
mysql  | 2023-05-04T18:28:09.899771Z 0 [ERROR] [MY-010020] [Server] Data Dictionary initialization failed.
```

Running a docker container on WSL ,after I removed the `.docker` file from the root with `sudo rm -r .docker` the problem was fixed.

**Comment by jose1894** [comment]
> 

It doesn't work on my server, fsutil return: It is not a directory

**Comment by dhanan-nd** [comment]
how to fix this error in ubuntu 23.04 , im having the same issue

**Comment by fractalzombie** [comment]
> > My solution:
> > cmd
> > fsutil file setCaseSensitiveInfo "C:\Users\XXXXX.laradock\data\mysql"
> > then
> > docker-compose up mysql
> 
> Thanks a lot works for windows 10. I have the same issue.

Yep, this thing is worked for me
